invisible_cities/cities/components.py

```diff
@@ -520,6 +521,8 @@ def cdst_from_files(paths: List[str]) -> Iterator[Dict[str,Union[pd.DataFrame, M
                 event_number, timestamp = evtinfo
                 yield dict(cdst    = cdst_df   .loc[cdst_df   .event==event_number],
                            summary = summary_df.loc[summary_df.event==event_number],
+                           kdst    = kdst_df   .loc[kdst_df   .event==event_number] if isinstance(kdst_df, pd.DataFrame) \
+                                                                                    else None,
```

Comment:
what do you think of this?

```suggestion
                this_event = lambda df: df.event==event_number
                yield dict(cdst    = cdst_df   .loc[this_event],
                           summary = summary_df.loc[this_event],
                           kdst    = kdst_df   .loc[this_event] if isinstance(kdst_df, pd.DataFrame) \
                                                                else None,
```

Reply:
I'm also thinking whether it is better to check `isinstance(kdst_df, pd.DataFrame)` or `kdst_df is not None`. The second one is perhaps more general, but I'm not convinced it's actually better. Thoughts?

Reply:
I like the `this_event` proposal, added. No opinion about the instance check.